packages/shared/lib/services/config.service.ts

```diff
@@ -114,7 +122,14 @@ class ConfigService {
     }
 
     async createProviderConfig(config: ProviderConfig): Promise<ProviderConfig | null> {
+        const provider = getProvider(config.provider);
+        if (!provider) {
+            throw new NangoError('unknown_provider');
+        }
```

Comment:
Is it possible (if not too annoying) to pass provider as params instead of loading it here? So it's side-effect "free"

Reply:
See f428e437